pkg/policy/selectorcache.go

```diff
@@ -37,11 +38,52 @@ type CachedSelector interface {
 	// be modified, as it is shared among multiple users.
 	GetSelections() []identity.NumericIdentity
 
+	// Selects return 'true' if the CachedSelector selects the given
+	// numeric identity.
+	Selects(nid identity.NumericIdentity) bool
+
+	// IsWildcard returns true if the endpoint selector selects
+	// all endpoints.
+	IsWildcard() bool
+
 	// String returns the string representation of this selector.
 	// Used as a map key.
 	String() string
 }
 
+// CachedSelectorSlice is a slice of CachedSelectors that can be sorted.
+type CachedSelectorSlice []CachedSelector
```

Comment:
This type introduction (and related changes elsewhere) could be split into a separate patch.

Reply:
Will consider tomorrow.